src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/removepolicy/TransportRemovePolicyAction.kt

```diff
@@ -0,0 +1,128 @@
+
+            val clusterStateRequest = ClusterStateRequest()
+                    .clear()
+                    .indices(*request.indices.toTypedArray())
+                    .metadata(true)
+                    .local(false)
+                    .indicesOptions(strictExpandOptions)
+
+            client.admin()
+                    .cluster()
+                    .state(clusterStateRequest, object : ActionListener<ClusterStateResponse> {
+                        override fun onResponse(response: ClusterStateResponse) {
+                            processResponse(response)
+                        }
+
+                        override fun onFailure(t: Exception) {
+                            actionListener.onFailure(t)
+                        }
+                    })
+        }
+
+        @Suppress("SpreadOperator") // There is no way around dealing with java vararg without spread operator.
+        fun processResponse(clusterStateResponse: ClusterStateResponse) {
+            val state = clusterStateResponse.state
+            populateLists(state)
+
+            if (indicesToRemovePolicyFrom.isNotEmpty()) {
+                val updateSettingsRequest = UpdateSettingsRequest()
+                        .indices(*indicesToRemovePolicyFrom.map { it.name }.toTypedArray())
+                        .settings(Settings.builder().putNull(ManagedIndexSettings.POLICY_ID.key))
+
+                try {
+                    client.execute(UpdateSettingsAction.INSTANCE, updateSettingsRequest,
+                            object : ActionListener<AcknowledgedResponse> {
+                                override fun onResponse(response: AcknowledgedResponse) {
+                                    if (response.isAcknowledged) {
+                                        updated = indicesToRemovePolicyFrom.size
+                                    } else {
+                                        updated = 0
+                                        failedIndices.addAll(indicesToRemovePolicyFrom.map {
+                                            FailedIndex(it.name, it.uuid, "Failed to remove policy")
+                                        })
+                                    }
+
+                                    actionListener.onResponse(RemovePolicyResponse(updated, failedIndices))
+                                }
+
+                                override fun onFailure(t: Exception) {
+                                    actionListener.onFailure(t)
+                                }
+                            }
+                    )
+                } catch (e: ClusterBlockException) {
+                    failedIndices.addAll(indicesToRemovePolicyFrom.map {
+                        FailedIndex(it.name, it.uuid, "Failed to remove policy due to ClusterBlockException: ${e.message}")
+                    })
+                    updated = 0
+                    actionListener.onResponse(RemovePolicyResponse(updated, failedIndices))
```

Comment:
This needs to be onFailure ? Whats the return HTTP code for this case?

Reply:
previously we use OK(200) as return code [link](https://github.com/opendistro-for-elasticsearch/index-management/blob/master/src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/resthandler/RestRemovePolicyAction.kt#L129)